### PR TITLE
Query schema by did

### PR DESCRIPTION
**Pull request by mcjcloud**
Query schemas by DID alone.

## Changes

- Add WhatIsByDid method to chain
- Return empty list from SchemaByCreator instead of nil

<sub> Mention [stepsize] in a comment if you'd like to report some technical debt. See examples [here](https://app.stepsize.com/api/demo-pr-redirect).</sub>

### CODE REVIEW

File: x/schema/keeper/schema.go [hunk 1]
```diff
@@ -119,6 +119,26 @@ func (k Keeper) GetWhatIsFromCreator(ctx sdk.Context, creator string) (val []*ty
...
+
+	for ; iterator.Valid(); iterator.Next() {
+		instance := &types.WhatIs{}
+		if err := k.cdc.Unmarshal(iterator.Value(), instance); err != nil {
+			return nil, false
+		}
+		if instance.Did == did {
+			return instance, true
+		}
+	}
```
Comment (joshLong145) [inline x/schema/keeper/schema.go]: I could be wrong but i dont think you need to iterate over the store, i think you can get the 'WhatIs' from the keeper of the 'DID' as that is they key used for content within the schema keeper.

Comment (mcjcloud) [reply x/schema/keeper/schema.go]: Good catch, fixed.

**Review by joshLong145 — APPROVED** [review]

**Comment by ntindle** [comment]
Does this change things for speedway

**Review by ntindle — APPROVED** [review]

**Review by joshLong145 — APPROVED** [review]

**Comment by mcjcloud** [comment]
No changes for speedway @ntindle, unless you want to add a similar query method

**Pull request merged**